src/composables/map/ol.synchronizer.ts

```diff
@@ -58,7 +63,36 @@ export class OlSynchronizer {
 
     watch(
       () => mapStore.bgLayer,
-      bgLayer => bgLayer !== undefined && openLayers.setBgLayer(map, bgLayer)
+      bgLayer =>
+        bgLayer !== undefined &&
+        openLayers.setBgLayer(map, bgLayer, styleStore.bgVectorSources)
+    )
+
+    watch(
+      () => styleStore.bgStyle,
+      newStyle =>
+        openLayers.applyOnBgLayer(map, bgLayer =>
+          styleService.applyStyle(bgLayer, newStyle || [])
+        )
+    )
+
+    watch(
+      () => styleStore.bgVectorSources,
+      newVectorSources => {
+        for (const id in newVectorSources) {
+          if (
+            !this.previousVectorSources ||
+            this.previousVectorSources[id] != newVectorSources[id]
+          ) {
+            openLayers.removeFromCache(id)
+            if (id == mapStore?.bgLayer?.id) {
```

Comment:
`===`

Reply:
I still don't get why you need all this casting. 
```
export interface VectorSourceDict {
  [id: LayerId]: IMvtConfig
}
```
Maybe this would work?

Reply:
https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCoAsIFsIDkD2AJhALJEQA2yA3gFDLLCEBcyIArlgEbT1tw5WAZzBRQAcz7AE+EAH5hoiXwTpgFQlAjzWGbHnJliFANoBdWgF9atGSBHIwmQWmcHiRysgC8NKS2QARgAaPhABCFYAciw4UCjQhmlZaNj4xORVdU1tVhM+BjoGYsYAgCYMkvCXKK44IWAhYhAEguKsjS0QPLaSopKBplYAZkqBhmrI5Ci4QixQRtE4MGAANwgAfQhQTGBIFrGS63GzMePkC2taGHYQBBXZZBhQQgAhAE8ASUIACn9WAAycHe0G+GRA5AUrn0BA85AotAAlLo3LDSPDkAAfZC3YjPEAQQh+MLkHxsUmY7FOfRSGDIH4Q4hyAB0TB83l8TERfhKWjA7CgIHJxD4lmQlCEKGAdIZkOZHRyIG5-QYMHwUHpdgcCuQ+DpjIg8rUnW0yt6mVkDliYFUZPxby+vyYwUyxsIiPN0vp1tUZvGDD5AqFPvQ5vODHO50DgqsNjA7wADiggSCoN8yRxuNAscgRGIQJJaKBILBEChPiRVmAAMKyZ7iHnIChwHgURT5yQMETvChTPPKK7F6DwJDIADKSgLABFpGBGyYhrnJ+IzKwK1XayB67Gi+Bh2XkCnoDP7vPF0e04RV8h1zW68BJFctXPKIFAcDQUTfAAiQLf2yWi+FBlKwmY8BqvhlAB9j4L2zIUPg4g-CYr4upQZRmB6WqwYaCFIfaHzfD8r6Ilhlo4fBiE-ARjrEcBpE2M+yCEG+47Liec6+NQVwsShFCBGYZLUM2rasN+-Hfi63a9mJQh-tYAD0ClsJw4HIIgSAJpARL1MgADWEDvOK6xCl6EJzkISYINKwCEowQqoGOuZWbQvHoYJXEiZQYnAZJS49lM35CGU35PoBzEgYeH5QBxQk8WUfECUJXltsg4l-lJ8YyWlcmha5CXuclLbeWlvmZQFskhWF9hzoQwysNQC4BBe6bYkI7zcLB163pu9Zilx1hMYQAAsrAkHACYADwtYQLo9fe4gAHwZhAADuyDjQmPwetBDiUMMZJjh1XCwT8ww7XViUeTQKU+Rl-nZUF8m0EpubHbB5JzhpEBaXZulwPphk4iAFDAAZyCrWIYD7PZzH4AgrnDHxwzXcJxWpeJwx+dJgVCFj1gjcykpgHRIQ3ejd3Y1luPyR6r3xkmQjIAmWiSlA6yEK5w1ExAJPoS6aOiaVIXlY9wWhTt2FwXhyFRAZ7xM6AzFvgkyAAPJcAAVhA9zMvLQg-CxiKYbtFEyyYcuGYrQqECBqsa9ruv64bZTG2RMHS1RFv67DdXMPbWs62AetW4b50m1LuFe5bCu+6NqsAIJQFAwLMjAUD4Fghvc87pEm7uJYjigeg4LF-RNe+qbfNeJfuOixixkNABsKL6LFA27bVADsY0TdN0VgtCOBop4FDLb4BLrZt22MeFzapkzvgmGTFTICj1zqpq4VsnqTbRUIfoWjVwooL4NFEVyfCvZHlFIQaHpJDK9+NgwhBN0110Gm0hBdzzJPOifB+yBrDWDVBqH4TE2RKzfofJiBo7QvEIk6d0V8FI3xlvfFQ5FPZISai6AABgAEmoEwMUxCDRyCiFEf25lpiWDVLiX2Td8Em1AZvCB28iS7x-iHBW21YHhXgWfRBtFL4MCUugqimCGCSNwQAohJDCBkOoBQqhND8BziiPQ-AjDoFdxYR6RSykGbJgHkSYAWAEyg2smACgRk7DrCgJARw+AlwdlhvrU2ODZY+2gS3AOjtg7OxgSbemiZTFVyJCzCAbMOaw02l4qOuCY7W2Yj3ROydU7p0zobX+uc3ZAA

Reply:
Will change to Map instead of native dict, which can handle correctly non-string types as keys